apiserver/facades/client/secrets/secrets.go

```diff
@@ -338,7 +349,93 @@ func fromUpsertParams(p params.UpsertSecretArg) state.UpdateSecretParams {
 	}
 }
 
-// CreateSecrets isn't on the v1 API.
+// UpdateSecrets isn't on the v1 API.
+func (s *SecretsAPIV1) UpdateSecrets(_ struct{}) {}
+
+// UpdateSecrets creates new secrets.
+func (s *SecretsAPI) UpdateSecrets(args params.UpdateUserSecretArgs) (params.ErrorResults, error) {
+	result := params.ErrorResults{
+		Results: make([]params.ErrorResult, len(args.Args)),
+	}
+	if err := s.checkCanAdmin(); err != nil {
+		return result, errors.Trace(err)
+	}
+	backend, err := s.getBackend(nil)
+	if err != nil {
+		return result, errors.Trace(err)
+	}
+	for i, arg := range args.Args {
+		err := s.updateSecret(backend, arg)
+		if errors.Is(err, state.LabelExists) {
+			err = errors.AlreadyExistsf("secret with label %q", *arg.Label)
+		}
+		result.Results[i].Error = apiservererrors.ServerError(err)
+	}
+	return result, nil
+}
+
+func (s *SecretsAPI) updateSecret(backend provider.SecretsBackend, arg params.UpdateUserSecretArg) (errOut error) {
+	uri, err := coresecrets.ParseURI(arg.URI)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	if arg.AutoPrune == nil && arg.Description == nil && arg.Label == nil && len(arg.Content.Data) == 0 {
```

Comment:
This should be a Validate method on the args struct like we do elsewhere (I think for charm secrets)